JavaFXUserInterface/src/main/java/controllers/AssetsController.java

```diff
@@ -190,99 +213,162 @@ private void sortingSetUp() {
         });
     }
 
+    /**
+     * Live search for assets in the assets list based on the input from the search bar.
+     * When a search has a match, all systems panes are cleared and generated again with the appropriate assets.
+     * When a search has no match, all system panes are cleared effectively showing no assets.
+     *
+     * @author Najim
+     */
+    private void searchAssets() {
+        searchedAssets = FXCollections.observableArrayList();
+        search.textProperty().addListener((observable, oldValue, newValue) -> {
+            searchedAssets.clear();
+            if (!newValue.equals("")) {
+                assets.stream().filter(o -> StringUtils.containsIgnoreCase(o.getRul().get(), newValue) || StringUtils.containsIgnoreCase(o.getName(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getDescription(), newValue) || StringUtils.containsIgnoreCase(o.getSerialNo(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getManufacturer(), newValue) || StringUtils.containsIgnoreCase(o.getCategory(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getSite(), newValue) || StringUtils.containsIgnoreCase(o.getLocation(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getRecommendation(), newValue) || StringUtils.containsIgnoreCase(assetTypeDAO.getNameFromID(o.getAssetTypeID()), newValue)).forEach(
+                        searchedAssets::add //Assets matched with search are added to the searchAssets list
+                );
+                if (searchedAssets.isEmpty()) {
+                    searchMatch = "No Match";
+                } else {
+                    searchMatch = "Match";
+                }
+            } else {
+                searchMatch = "No Search";
+            }
+
+            if (assetsTabPane.getSelectionModel().getSelectedItem().getId().equals("thumbnailTab")) {
+                assetsThumbPane.getChildren().clear();
+                generateThumbnails();
+            } else {
```

Comment:
This is really cherry-picking, but I think that having else if with the list tab specified instead of else might provide more scalability in case there are future uses and addition of other tabs. It also clarifies for developers the intention of this conditional statement (that the following code is meant for a specific tab as well, not for all other possibilities). You can let me know what you think.

Reply:
Great suggestion. The way you explained it makes more sense to use else if